File: web/lib/api/conversation-delegation.ts
import { request } from "@/lib/api/client"

export type DelegationView = {
  conversationId: number; ownerId: number; ownerName: string; canManage: boolean
  active: boolean; previewOnly: boolean; revision: number; aiAgentId: number; instructions: string
  startedByName: string; startedAt?: string; expiresAt?: string; endReason: string
  agents: { id: number; name: string; liveAvailable: boolean }[]
  events: { id: number; actorName: string; sourceMessageId: number; kind: string; reason: string; content: string; previewOnly: boolean; createdAt: string }[]
}
export type DelegationInput = { revision: number; aiAgentId: number; previewOnly: boolean; durationMinutes: number; instructions: string }
export function fetchDelegation(id: number) { return request<DelegationView>(`/api/dashboard/conversation/${id}/delegation`) }
export function startDelegation(id: number, body: DelegationInput) {
  return request<void>(`/api/dashboard/conversation/${id}/delegation/start`, { method: "POST", body: JSON.stringify(body) })
}
export function stopDelegation(id: number, revision: number) {
  return request<void>(`/api/dashboard/conversation/${id}/delegation/stop`, { method: "POST", body: JSON.stringify({ revision }) })
}
export function previewDelegation(id: number, revision: number) {
  return request<void>(`/api/dashboard/conversation/${id}/delegation/preview`, { method: "POST", body: JSON.stringify({ revision }) })
}

File: internal/pkg/dto/response/conversation_delegation.go
package response

import "time"

type DelegationAgentOption struct {
	ID            int64  `json:"id"`
	Name          string `json:"name"`
	LiveAvailable bool   `json:"liveAvailable"`
}

type ConversationDelegationView struct {
	ConversationID int64                         `json:"conversationId"`
	OwnerID        int64                         `json:"ownerId"`
	OwnerName      string                        `json:"ownerName"`
	CanManage      bool                          `json:"canManage"`
	Active         bool                          `json:"active"`
	PreviewOnly    bool                          `json:"previewOnly"`
	Revision       int64                         `json:"revision"`
	AIAgentID      int64                         `json:"aiAgentId"`
	Instructions   string                        `json:"instructions"`
	StartedByName  string                        `json:"startedByName"`
	StartedAt      *time.Time                    `json:"startedAt"`
	ExpiresAt      *time.Time                    `json:"expiresAt"`
	EndReason      string                        `json:"endReason"`
	Agents         []DelegationAgentOption       `json:"agents"`
	Events         []ConversationDelegationEvent `json:"events"`
}

type ConversationDelegationEvent struct {
	ID              int64     `json:"id"`
	ActorName       string    `json:"actorName"`
	SourceMessageID int64     `json:"sourceMessageId"`
	Kind            string    `json:"kind"`
	Reason          string    `json:"reason"`
	Content         string    `json:"content"`
	PreviewOnly     bool      `json:"previewOnly"`
	CreatedAt       time.Time `json:"createdAt"`
}
